Quran/AyahInfoPersistenceStorage.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  AyahInfoPersistenceStorage.swift
+//  Quran
+//
+//  Created by Ahmed El-Helw on 5/12/16.
+//  Copyright © 2016 Quran.com. All rights reserved.
+//
+
+import Foundation
+import SQLite
+
+struct AyahInfoPersistenceStorage {
```

Comment:
My first intention was having an abstraction layer to abstract the SQLite library or let's say, driver. So that we can replace it later with minimal changes. Then, we will have an Ayah Info class that will use the abstraction layer. It will be the same for Gapless Qari sqlite database.

So, `SQLitePersistenceStorage` should be a protocol that could have one implementation with sqlite.swift then, in the future if we find another library we could use it then all we need to do is just implement SQLitePersistenceStorage and update the dependency injection container to use the new one. Feel free to modify SQLitePersistenceStorage protocol methods signatures that best suit our needs. I think it could be abstracted better like having a similar API as Sqlite.swift has.

Then we could have AyahInfoRetriever protocol that does what you said like `getAyahInfoForAyah`, `getAyahInfoForPage`, and `getAyahInfoForXY`. Then the one implementation called `SQLiteAyahInfoRetriever` that uses `SQLitePersistenceStorage`.

Then we will have another one for Qari database like `QariTimingRetriever` and `SqliteQariTimingRetriever`. 

But I'm also totally fine with your approach.